pageserver/src/tenant.rs

```diff
@@ -2547,35 +2546,21 @@ fn run_initdb(
     initdb_target_dir: &Path,
     pg_version: u32,
 ) -> anyhow::Result<()> {
-    let initdb_bin_path = conf.pg_bin_dir(pg_version)?.join("initdb");
-    let initdb_lib_dir = conf.pg_lib_dir(pg_version)?;
-    info!(
-        "running {} in {}, libdir: {}",
-        initdb_bin_path.display(),
-        initdb_target_dir.display(),
-        initdb_lib_dir.display(),
-    );
-
-    let initdb_output = Command::new(&initdb_bin_path)
-        .args(["-D", &initdb_target_dir.to_string_lossy()])
-        .args(["-U", &conf.superuser])
-        .args(["-E", "utf8"])
-        .arg("--no-instructions")
-        // This is only used for a temporary installation that is deleted shortly after,
-        // so no need to fsync it
-        .arg("--no-sync")
-        .env_clear()
-        .env("LD_LIBRARY_PATH", &initdb_lib_dir)
-        .env("DYLD_LIBRARY_PATH", &initdb_lib_dir)
-        .stdout(Stdio::null())
```

Comment:
Why are we not using this for all callers?

Reply:
Technically, there's some useful stdout that might lead us to thinking:

```
~/Downloads ❯ initdb ./zzz/ 1>stdout.log 2>stderr.log

~/Downloads ❯ cat stdout.log
The files belonging to this database system will be owned by user "someonetoignore".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.UTF-8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

creating directory ./zzz ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... Europe/Helsinki
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok


Success. You can now start the database server using:

    pg_ctl -D ./zzz/ -l logfile start


~/Downloads ❯ cat stderr.log
initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
```

For instance, I find that Helsinki TimeZone even in the programmatic initdb call logs, that could be also pinned instead.
So, I'd leave stdout enabled in the library method: if some tools are not interested in that, `Command` api could be used for disabling.